pyflakes/test/test_api.py

```diff
@@ -648,6 +664,43 @@ def test_checkRecursive(self):
         finally:
             shutil.rmtree(tempdir)
 
+    def test_stdinReportsErrors(self):
+        """
+        L{check} reports syntax errors from stdin
+        """
+        source = "max(1 for i in range(10), key=lambda x: x+1)\n"
+        err = io.StringIO()
+        count = withStderrTo(err, check, source, "<stdin>")
+        self.assertEqual(count, 1)
+        errlines = err.getvalue().split("\n")[:-1]
```

Comment:
I don't think this actually tests the behaviour -- and we already have a SyntaxError test

Reply:
The behavior is triggered when ast.parse throws a SyntaxError from code with a filepath that doesn't exist (stdin).

I'll see if I can trigger this with a different syntax error that has more consistent messaging to avoid the rats nest of expected messages by version. The offset behavior is a bit concerning though. It looks like pypy has special handling that is objectively wrong in certain situations (such as this one).

Reply:
pypy is usually pretty attentive to fixes for syntax error offsets (with a little help from me they inspired the improvements to cpython 3.10's error messages)

Reply:
I meant pyflakes handling of pypy's offsets as here: https://github.com/PyCQA/pyflakes/blob/44ef321b0e608c61182ecf5d88f9edeececa5403/pyflakes/api.py#L46-L56

It's probably something that used to be an issue with pypy and was fixed since on the other end. I tested with pypy3.7-7.3.9, but tracking down which version of pypy changed offsets in syntax errors is out of scope for here.

Reply:
oh yeah we should totally kill that -- forgot pyflakes had special cases (probably from like 2015-era pypy)